### ISSUE TITLE
FTM session to ASAP-capable FTMR crashes (IDFGH-6470)

### ISSUE THREAD

**Issue by crackwitz** [espressif/esp-idf#8127]
I have two ESP32-S2. I also have a Linksys velop AP that is an FTMR but doesn't advertise it. It's also **ASAP-capable**.

Using the `ftm` example in esp-idf and FTMing to that velop AP, it succeeds with `non-ASAP` some times, but other times fails and crashes when the log indicates `ASAP`.

```
ftm> ftm -s "Velop" # -c 32 behaves the same
I (29606) ftm_station: Requesting FTM session with 32 frames, burst 200 ms (0: No Pref)
W (29606) wifi:Starting FTM session with e8:9f:80:e5:aa:1b in 3.600 Sec
W (29606) wifi:Mode: non-ASAP, Bursts: 8, FTM's per burst: 4, Burst Period: 900mSec, Burst Duration: 32000uSec
W (38616) wifi:FTM session ends with 23 valid readings out of 31, Avg raw RTT: 253.322 nSec, Avg RSSI: -47
```
(... but even that fails sometimes, with `E (441856) wifi:No valid FTM Measurements found!` or with `W (450486) wifi:Start time delay for FTM too high! (4231.458 Sec)` or simply with status 2 (CONF_REJECTED))

`-c 16` and `-c 24` succeed too (or say "no measurements"), but only when `non-ASAP` is reported.

This crashes completely:
```
ftm> ftm -s "Velop" -c 8
I (23086) ftm_station: Requesting FTM session with 8 frames, burst 200 ms (0: No Pref)
W (23086) wifi:Starting FTM session with e8:9f:80:e5:aa:1b
W (23086) wifi:Mode: ASAP, Bursts: 1, FTM's per burst: 8, Burst Period: 200mSec, Burst Duration: 0uSec
E (28096) task_wdt: Task watchdog got triggered. The following tasks did not reset the watchdog in time:
E (28096) task_wdt:  - IDLE (CPU 0)
E (28096) task_wdt: Tasks currently running:
E (28096) task_wdt: CPU 0: wifi
E (28096) task_wdt: Print CPU 0 (current core) backtrace


Backtrace:0x40094FE7:0x3FFCC9D00x4002532D:0x3FFCCA00 0x400DACA9:0x3FFDDCE0 0x400D9476:0x3FFDDD00 0x40039C9B:0x3FFDDD30 0x4003A162:0x3FFDDD50 0x4003A2C5:0x3FFDDD90 0x40038262:0x3FFDDDD0 0x40031C12:0x3FFDDE00
```

Additionally, in my own custom code, where the ESP32-S2 runs in APSTA mode, with `{ .frm_count = 32, .burst_period = 2 }`, I get this:
```
W (7225) wifi:Starting FTM session with e8:9f:80:e5:aa:1b
W (7225) wifi:Mode: ASAP, Bursts: 1, FTM's per burst: 8, Burst Period: 0mSec, Burst Duration: 0uSec
```

# Environment

- Development Kit:      ESP32-S2-DevKitM (v1 I guess)
- Module or chip used:  ESP32-S2-MINI-1
- IDF version: `v5.0-dev-810-gb886dc6998` (from 2021-12-21)
- Build System:         idf.py / vscode extension
- Compiler version: `xtensa-esp32s2-elf-gcc (crosstool-NG esp-2021r2) 8.4.0`

**Comment by nachiketkukade** [comment]
@crackwitz , thanks for bringing this issue to our attention.
First a question on the APSTA mode, are you seeing the same failure with another ESP32S2?
Coming back to the issue, there are 3 different ones at play here -
1. Watchdog timeout (crash you mentioned)
2. Consistent failure in ASAP mode
3. Intermittent failures in non-ASAP mode
The first 2 issues are being addressed with a fix. But the 3rd one cannot be fixed entirely from our side. The reason being that the Linksys Velop behaves erroneously and sends multiple frames with no measurement in every iteration. It also terminates the session abruptly sometimes. This is why the 'No measurements' error keeps coming. This issue is still present in ASAP mode but to a lesser extent. I've tested an Mi11X Pro Smartphone with the Velop AP using the RTT app which also uses ASAP mode, it too reports around 30% failure rate. Once the fix for the first 2 issues is merged, do let us know your observations on the failures in ASAP mode with this AP. Thanks.

**Comment by crackwitz** [comment]
Thank you! I'll try the master branch right now.

> First a question on the APSTA mode, are you seeing the same failure with another ESP32S2?

I have several ESP32-S2-DevKitM and they behave interchangeably, when talking with the linksys velop. Among their own, all is fine.

I fully understand that Espressif isn't responsible for working around malfunctions in third party equipment like Linksys Velop. That device doesn't even advertise FTM. I don't expect it to work. What I am doing here could at best be considered "fuzz testing" or a very limited [plugfest](https://en.wikipedia.org/wiki/Plugtest). Intermittent "failures" of FTM sessions are perfectly acceptable to me, so long as the ESP side can't be crashed by a misbehaving FTMR on the other side :)

**Comment by AndreiKorchunov** [comment]
Board 
esp32 c3 devkit-02 both 
ESP-ROM:esp32c3-api1-20210207 Build:Feb  7 2021
Hardware Configuration
nothing special
Arduino IDE 1.8.19 (Windows Store 1.8.57.0)
Library esp32 v 2.0.6 - softAp not work by standart example for FTM. The access point does not connect to the second esp 32, nor the Samsung Note. There are no errors in any esp32.
Library esp32 v 2.0.3 - softAp work by standart example for FTM. 
Intermittent crashes in ISAP mode FTM Error: CONF_REJECTED

**Comment by nachiketkukade** [comment]
@AndreiKorchunov , please create a new issue for this if you haven't already and share the logs
